app/open_ai_client.rb

```diff
@@ -1,111 +1,227 @@
   end
 
   def default_role_prompt
-<<-ROLE_PROMPT
-You are an advanced Teaching Assistant AI. Your task involves reviewing and providing feedback on student submissions.
-ROLE_PROMPT
+    <<~ROLE_PROMPT
+      You are an advanced Teaching Assistant AI. Your task involves reviewing and providing feedback on student submissions.
+    ROLE_PROMPT
   end
 
   def default_user_prompt
-<<-USER_PROMPT
-The student's submission will be as follows:
-${SUBMISSION}
-USER_PROMPT
+    <<~USER_PROMPT
+      The student's submission will be as follows:
+      ${SUBMISSION}
+    USER_PROMPT
   end
 
   def default_input_prompt
-<<-INPUT_PROMPT
-The student's submissions will be an array of objects following the provided schema:
-
-```json
-{
-  "kind": "The type of answer - can be shortText, longText, link, files, or multiChoice",
-  "title": "The question that was asked of the student",
-  "result": "The student's response",
-  "status": "Field for internal use; ignore this field during your review"
-}
-```
-INPUT_PROMPT
+    <<~INPUT_PROMPT
+      The student's submissions will be an array of objects following the provided schema:
+
+      {
+        "kind": "The type of answer - can be shortText, longText, link, files, or multiChoice",
+        "title": "The question that was asked of the student",
+        "result": "The student's response",
+        "status": "Field for internal use; ignore this field during your review"
+      }
+
+    INPUT_PROMPT
+  end
+
+  def default_evaluation_criteria_prompt
+    if ENV.fetch("ASSIGN_GRADES", "false") == "true"
+      <<~EC_PROMPT
+        The following is array of objects. Each object has two keys
+          - evaluation_criteria_id: This key stores the identifier for the evaluation criteria, which can be either a numeric value or a string. This identifier is unique for each set of criteria and is used to reference the specific evaluation criteria being described.
+          - allowed_grades": Associated with this key is an array of integers, which represents the set of permissible grades for associated evaluation criterion(evaluation_criteria_id). These grades are predefined and indicate the possible outcomes or ratings that can be assigned based on the evaluation criterion.
+
+          The evaluation_criteria for this submission are:
+            ${SUBMISSION_EC}
+      EC_PROMPT
+    else
+      ""
+    end
   end
 
   def default_output_prompt
-<<-OUTPUT_PROMPT
-Please provide your response in the following JSON format. Adhere to the format strictly and escape all line-breaks within strings using \\\\n.
-
-```json
-{
-    "status": "\"passed\" or \"failed\"",
-    "feedback": "Detailed feedback for the student in markdown format. Aim for a human-like explanation as much as possible."
-}
-```
-
-If the student submission is not related to question, share generic feedback.
-OUTPUT_PROMPT
+    <<~OUTPUT_PROMPT
+      The following is the expected json schema for the response.
+        #{function}
```

Comment:
Why are we adding the functions again in the prompt? This will increase the token size; OAI will pass on the function along with out prompt to the system. 

This looks redundant;

Reply:
Now, I had cleaned up the `output_prompt`